File: app/src/main/java/com/test/ashish/sendsmsnew/ViewModel.java
package com.test.ashish.sendsmsnew;

import android.app.Application;
import android.arch.lifecycle.AndroidViewModel;
import android.arch.lifecycle.MutableLiveData;
import android.support.annotation.NonNull;

import com.test.ashish.sendsmsnew.pojo.Contact;
import com.test.ashish.sendsmsnew.pojo.Message;

import java.util.ArrayList;


public class ViewModel extends AndroidViewModel {

    private ApiRepository apiRepository;
    private MutableLiveData<Boolean> contactApiStatus;
    private MutableLiveData<Boolean> messageApiStatus;
    private MutableLiveData<Boolean> updateMessageApiStatus;

    public ViewModel(@NonNull Application application) {
        super(application);
        apiRepository = new ApiRepository();
    }


    public MutableLiveData<ArrayList<Contact>> getContacts(){
        return  apiRepository.getAllContatcs();
    }

    public MutableLiveData<Boolean> getContactFetchStatus() {
        if (contactApiStatus == null) {
            contactApiStatus = apiRepository.getContactFetchStatus();
        }
        return contactApiStatus;
    }

    public MutableLiveData<ArrayList<Message>> getMessages(){
        return  apiRepository.getAllMessage();
    }

    public MutableLiveData<Boolean> getMessageFetchStatus() {
        if (messageApiStatus == null) {
            messageApiStatus = apiRepository.getMessageFetchStatus();
        }
        return messageApiStatus;
    }

    public MutableLiveData<Boolean> getUpdateMessageFetchStatus(){
        if (updateMessageApiStatus == null) {
            updateMessageApiStatus = apiRepository.getUpdateMessageFetchStatus();
        }
        return updateMessageApiStatus;
    }

    public void updateMessage(String useName, String userMobile, String userMessage){
        apiRepository.updateMessage(useName, userMobile, userMessage);
    }


}

File: app/src/main/java/com/test/ashish/sendsmsnew/ViewModel.kt
package com.test.ashish.sendsmsnew

import android.app.Application
import android.arch.lifecycle.AndroidViewModel
import android.arch.lifecycle.MutableLiveData

import com.test.ashish.sendsmsnew.pojo.Contact
import com.test.ashish.sendsmsnew.pojo.Message

import java.util.ArrayList


class ViewModel(application: Application) : AndroidViewModel(application) {

    private val apiRepository: ApiRepository
    private var contactApiStatus: MutableLiveData<Boolean>? = null
    private var messageApiStatus: MutableLiveData<Boolean>? = null
    private var updateMessageApiStatus: MutableLiveData<Boolean>? = null


    val contacts: MutableLiveData<ArrayList<Contact>>
        get() = apiRepository.allContatcs

    val contactFetchStatus: MutableLiveData<Boolean>
        get() {
            if (contactApiStatus == null) {
                contactApiStatus = apiRepository.contactFetchStatus
            }
            return contactApiStatus
        }

    val messages: MutableLiveData<ArrayList<Message>>
        get() = apiRepository.allMessage

    val messageFetchStatus: MutableLiveData<Boolean>
        get() {
            if (messageApiStatus == null) {
                messageApiStatus = apiRepository.messageFetchStatus
            }
            return messageApiStatus
        }

    val updateMessageFetchStatus: MutableLiveData<Boolean>
        get() {
            if (updateMessageApiStatus == null) {
                updateMessageApiStatus = apiRepository.updateMessageFetchStatus
            }
            return updateMessageApiStatus
        }

    init {
        apiRepository = ApiRepository()
    }

    fun updateMessage(useName: String, userMobile: String, userMessage: String) {
        apiRepository.updateMessage(useName, userMobile, userMessage)
    }


}